### ISSUE TITLE
Compliance/External user can view compliance obligation details

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-compliance#18]
#### Description:

As an external user, I can view the details of a compliance obligation (obligation summary).

**Top Header** Review [Reporting Year] Compliance Summary 

**Next Section** From [Reporting Year] Report
- [x] Emissions Attributable for Compliance [data point BCIERS Reporting]
- [x] Emissions Limit [data point BCIERS Reporting]
- [x] Excess Emissions [data point BCIERS Reporting]

**Next Section** [Reporting Year] Compliance Obligation
- [x] Obligation ID [data point BCIERS Compliance]
- [x] [Reporting Year] Compliance Charge Rate [data point BCIERS Reporting?]
- [x] Equivalent Value: Formula Compliance Obligation * Compliance Charge Rate

[Design in Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=32714-10350&m=dev)

#### Acceptance Criteria:

- Given that I am on my compliance obligations grid, when I click on manage compliance obligation, I am brought to the compliance obligation's detailed page
- Given that I'm on the compliance obligation summary, I am in the highlighted tab: Review 2024 Compliance Summary
- User should be able to see a page elements as described above
- Url should be `/compliance/compliance-summaries/{id}` 

Blocked by #133

**Comment by patrickisaac** [comment]
Refinement Notes:
- @nanyangpro to provide wireframes

**Comment by nanyangpro** [comment]
Link to design is updated in the ticket description

**Comment by vivanovbc** [comment]
@vesselak This ticket is similar to [this one](https://github.com/orgs/bcgov/projects/123/views/17?pane=issue&itemId=99736886&issue=bcgov%7Ccas-compliance%7C133). Could you review and consolidate them? Also, we should ensure consistent naming - it's referred to as 'Compliance Obligation Details' in one place and 'Compliance Report 2' in another.

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EQbS4gOXB9dLmjX0xk0zQbABMEe7FP-2cxSFgzu2j29-FA?e=Qcs67E).

I can verify that:
- [x] When a submitted report for an operation has excess emission, an obligation ID is created, and the excess emission is visible in the Compliance Summaries grid
- [x] When there is an excess emission, and user clicks `Manage Obligation`, the page routes to the Review Compliance Summary page for that operation
- [ ] When user is in the Compliance Summary page, the top header is `Review [Reporting Year] Compliance Summary` - **Note: Currently it says `Report Information`**
- [x] When user is in the Compliance Summary page, in the highlighted tab is `Review 2024 Compliance Summary`
- [ ] When user is in the Review Compliance Summary page, the url is `/compliance/compliance-summaries/{id}` - @vesselak @nataliepiguliak wanted to confirm that the correct URL should be `compliance/compliance-summaries/{id}/review-compliance-summary` once user clicks `Manage Obligation`?
- [x] When user is in the Review Compliance Summary page, the operation name is displayed at the top right as the heading
- [ ] When user is in the Review Compliance Summary page, the values `Emissions Attributable for Compliance`, `Emissions Limit`, and `Excess Emissions` are pulled from the Compliance Summary in the operation's submitted report - **Note: The three fields are not formatted in 4 decimal places, the value for `Excess Emission` is also rounded off which affects the `Equivalent Value` field**
- [ ] When user is in the Review Compliance Summary page, under [Reporting Year] Compliance Obligation section, the value for fields `Obligation ID`, `[Year] Compliance Charge Rate`, and `Equivalent Value` are correct - **Note: Value for Compliance Charge Rate is correct but uses incorrect unit (should be $ /tCO2e), and unit for Equivalent Value is also incorrect (should be $, with 2 decimal places)**
- [ ] When user is in the Review Compliance Summary page, under Outstanding Compliance Obligation section, value for `Outstanding Balance` is undefined, 

@vesselak, @nataliepiguliak please see the Notes I've added on the boxes. I do have #170 currently open which I can update to have these issues tackled. Please let me know if that would be sufficient. Moving to Pending PO for review of test and notes. Thank you!

**Comment by vesselak** [comment]
@nataliepiguliak  Would it be possible to fix those three as part of this ticket and have the rest move to #170 as @fviduya suggested?

- [ ] When user is in the Review Compliance Summary page, the values Emissions Attributable for Compliance, Emissions Limit, and Excess Emissions are pulled from the Compliance Summary in the operation's submitted report - **Note: The three fields are not formatted in 4 decimal places, the value for Excess Emission is also rounded off which affects the Equivalent Value field**

- [ ] When user is in the Review Compliance Summary page, under [Reporting Year] Compliance Obligation section, the value for fields Obligation ID, [Year] Compliance Charge Rate, and Equivalent Value are correct - **Note: Value for Compliance Charge Rate is correct but uses incorrect unit (should be $ /tCO2e), and unit for Equivalent Value is also incorrect (should be $, with 2 decimal places)**

- [ ] When user is in the Review Compliance Summary page, under Outstanding Compliance Obligation section, value for Outstanding Balance is undefined,

**Comment by vesselak** [comment]
@fviduya based on Natalie's update this morning she is working on the 3 bullets so I moved the ticket to In Progress.
cc @nataliepiguliak

**Comment by fviduya** [comment]
Thanks @vesselak, I found that the first unchecked box is already included in #170.

**Comment by fviduya** [comment]
I have retested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EQbS4gOXB9dLmjX0xk0zQbABMEe7FP-2cxSFgzu2j29-FA?e=D5rHCx&nav=eyJoIjoiMjk0MjM4NjM1In0).

I can verify that:
- [ ] When user is in the Review Compliance Summary page, the values Emissions Attributable for Compliance, Emissions Limit, and Excess Emissions are pulled from the Compliance Summary in the operation's submitted report - **Note: The value for Excess Emission is still rounded off which affects the Equivalent Value field**
- [ ] When user is in the Review Compliance Summary page, under [Reporting Year] Compliance Obligation section, the value for fields Obligation ID, [Year] Compliance Charge Rate, and Equivalent Value are correct - **Note: the units are already correct, but the Equivalent Value is affected by the Excess Emissions field being rounded off)
- [ ] When user is in the Review Compliance Summary page, under Outstanding Compliance Obligation section, value for Outstanding Balance is pulled from Compliance Summary page - **Note: same as the first box, the value here is rounded off as well**

Hi @nataliepiguliak, I'm moving this back to In Progress for the changes regarding rounding off which seem to affect the others as well. cc: @vesselak

**Comment by fviduya** [comment]
I have retested this in the DEV environment and the issue regarding Excess Emission being rounded off still persists, I'm moving this to Pending PO and will create a separate ticket for this. [Here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EQbS4gOXB9dLmjX0xk0zQbABMEe7FP-2cxSFgzu2j29-FA?e=tJQDTN&nav=eyJoIjoiMzYyMzM1NTA5In0%3D).

**Comment by vesselak** [comment]
Reviewed the QA documentation and in dev. Accepting ticket with the deficiencies documented in #170 and the rounding errors to be addressed in separate ticket. 
@fviduya please add the ticket number next week when time permits.
